File: solutions/2029-stone-game-ix/greedy.js
// time: O(n)
// space: O(1)

/**
 * @param {number[]} stones
 * @return {boolean}
 */
const stoneGameIX = stones => {
    const mod = [0, 0, 0];

    for (const stone of stones) mod[stone % 3]++;
    if (mod[0] % 2 === 0) return mod[1] > 0 && mod[2] > 0;

    return Math.abs(mod[1] - mod[2]) > 2;
};

File: solutions/2029-stone-game-ix/greedy.cpp
#include <cstdlib>
#include <vector>
using namespace std;

class Solution {
  public:
    bool stoneGameIX(vector<int> stones) {
        int mod[3] = {};

        for (const int num : stones) mod[num % 3]++;
        if (mod[0] % 2 == 0) return mod[1] > 0 && mod[2] > 0;

        return abs(mod[1] - mod[2]) > 2;
    }
};
